ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -108,11 +106,13 @@ public NetworkClient(Selector selector, NetworkConfig networkConfig, NetworkMetr
       List<NetworkSend> sends = prepareSends(responseInfoList);
       selector.poll(pollTimeoutMs, sends);
       handleSelectorEvents(responseInfoList);
-      return responseInfoList;
+    } catch (Exception e) {
+      networkMetrics.networkClientException.inc();
```

Comment:
Will log the error. Handling it here, because as per the abstraction that I want to keep, `NetworkClient` is the caller.